src/text.c

```diff
@@ -363,6 +372,11 @@ int main( int argc, char *argv[] )
 			sprintf( string, "%s.st", axel->filename );
 			if( access( axel->filename, F_OK ) == 0 )
 			{
+				if( conf->no_clobber && access( string, F_OK ) != 0 )
+				{
+					printf("File %s exists\n", axel->filename );
+					goto close_axel;
+				}
```

Comment:
Instead of doing the check here twice, why not just doing it in axel_new() right after having assigned axel->filename?
Don't forget that in axel_new() a first connection is already made to retrieve the file size. By doing the check there you have a chance to prevent such connection (it is actually useless if we already know that the file won't be downloaded).

Reply:
I just attempted something and the result is also much simpler. I'd really recommend looking into axel_new().

Reply:
+1.